java/arcs/sdk/testing/BaseTestHarness.kt

```diff
@@ -0,0 +1,92 @@
+package arcs.sdk.testing
+
+import arcs.core.data.HandleMode
+import arcs.core.host.EntityHandleManager
+import arcs.core.storage.driver.RamDisk
+import arcs.core.storage.driver.RamDiskDriverProvider
+import arcs.core.storage.handle.HandleManager
+import arcs.core.storage.keys.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.jvm.util.JvmTime
+import arcs.sdk.EntitySpec
+import arcs.sdk.Handle
+import arcs.sdk.Particle
+import arcs.sdk.testing.HandleFlavor.COLLECTION
+import arcs.sdk.testing.HandleFlavor.SINGLETON
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Indicates the flavor of a handle: Singleton or a Collection (more may come).
+ */
+enum class HandleFlavor {
+    SINGLETON,
+    COLLECTION
+}
+
+/**
+ * Groups all the per-handle information that's relevant to the TestHarness.
+ */
+data class HandleDescriptor(
+    val name: String,
+    val entitySpec: EntitySpec<*>,
+    val flavor: HandleFlavor
+)
+
+/**
+ * A base class for Test Harnesses for Kotlin particles.
+ */
+open class BaseTestHarness<P : Particle>(
+    private val factory: () -> P,
+    private val descriptors: List<HandleDescriptor>,
+    scope: CoroutineScope
+) {
+    protected val handles: MutableMap<String, Handle> = mutableMapOf()
+    lateinit var particle: P
```

Comment:
if this is going to be public, some kdoc would be good for it - to explain why it's lateinit, and what needs to be called to ensure it has been initialized, etc.